Add customCloseIcon prop

moonboots opened the pull request:
@majapw @ljharb 

Adds a `customCloseIcon` prop to allow overriding the default `<CloseButton />`. I matched the behavior of `customArrowIcon`: default prop of null, fallback to original component inside `render`, and a wrapping css class.

majapw commented on src/components/DateRangePicker.jsx:
```diff
@@ -273,7 +277,9 @@ export default class DateRangePicker extends React.Component {
               {this.props.phrases.closeDatePicker}
             </span>
-            <CloseButton />
+            <div className="DateRangePickerInput__close">
```
This class name is wrong. You're not inside the input.

majapw commented on stories/DateRangePicker_input.js:
```diff
@@ -30,6 +30,19 @@ const TestCustomArrowIcon = () => (
+      padding: '3px',
+    }}
+  >
+    {'X'}
```
This can just be `X` no?

majapw commented on test/components/DateRangePickerInput_spec.jsx:
```diff
@@ -117,6 +117,17 @@ describe('DateRangePickerInput', () => {
+          showClearDates
+          customCloseIcon={<span className="custom-close-icon" />}
+        />);
+      expect(wrapper.find('.DateRangePickerInput__calendar-icon .custom-close-icon'));
```
This should fail... why would it be under `.DateRangePickerInput__calendar-icon`? Why isn't this failing?

moonboots commented:
@majapw I've added a corresponding spec for SingleDatePicker. The spec you pointed out wasn't failing because it accidentally didn't have any assertions, only a dangling `expect(..);`, fixed now.

majapw commented on test/components/SingleDatePickerInput_spec.jsx:
```diff
@@ -54,6 +54,17 @@ describe('SingleDatePickerInput', () => {
+            showClearDates
+            customCloseIcon={<span className="custom-close-icon" />}
+          />);
+        expect(wrapper.find('.DateRangePickerInput__calendar-icon .custom-close-icon'));
```
You now have the same problem here in this spec...

moonboots replied:
Oops, fixed

zeemawn commented:
Correct me if I'm wrong, but shouldn't this whole thing rather be called "custom CLEAR icon"?
The 'X' might look like a close-icon, but its function is to clear the dates, isn't it?
(Besides that, it would be nice to have an actual close-button)

ljharb commented:
It closes the datepicker, it doesn't clear the dates.

majapw commented:
It actually does both... We overload the icon in use, so uh, I get the Q

ljharb commented:
Lol well there you go

ljharb commented:
@majapw can we unoverload them?

interactivejohn commented:
I ran into the same issue as @zeemawn. I spent a good couple hours thinking that I could add a custom close icon to the default state, only to find out that it was only implemented for the `showClearDate` and `withFullScreenPortal` props. Maybe something can be added to the README that explains where the `customCloseIcon` is implemented? Meanwhile I will fork this package and start working on implementing a close functionality for the default `DatePicker` as requested above (I also need it for a project).
